core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -2261,8 +2116,12 @@ case object TopicDeletion extends ControllerEvent {
   override def state: ControllerState = ControllerState.TopicDeletion
 }
 
-case class PartitionReassignment(reassignments: Option[Map[TopicPartition, Option[Seq[Int]]]],
-                                 callback: Option[AlterReassignmentsCallback]) extends ControllerEvent {
+case object ZkPartitionReassignment extends ControllerEvent {
+  override def state: ControllerState = ControllerState.AlterPartitionReassignment
```

Comment:
We probably want to use different states btw ZkPartitionReassignment and ApiPartitionReassignment?

Reply:
I thought the intent of `ControllerState` was to capture a higher level breakdown of where the controller was spending its time. For example, `AlterPartitionReassignment` state is used both for the initial reassignment trigger and for ISR state changes affecting a reassigning partition. It might be user to see a finer-grained breakdown based on the trigger, but I think practically speaking only one of them will be used in a given cluster.